### ISSUE TITLE
Black square between tabline and scrollbar

### ISSUE THREAD

**Issue by adinklotz** [macvim-dev/macvim#1295]
### Steps to reproduce

1. Run a config-less MacVim: `mvim -g --clean`
2. Set a non-black color scheme that lets us see what's happening: `:colorscheme slate`
3. Disable the GUI tabline - this only applies to the text tabline: `:set guioptions-=e`
4. Create a new tab so the tabline shows: `:tabnew`

<img width="823" alt="Screen Shot 2022-09-21 at 9 58 47 PM" src="https://user-images.githubusercontent.com/16441122/191641771-4f984855-d8fa-4f1d-bef3-29894c4d7a64.png">

Note the black space that appears at the top-right corner

### Expected behaviour

Any of the following:
1. The tabline extends all the way to the right edge
2. The scroll bar extends all the way up to the title bar
3. That gap is left empty but matches the colorscheme background color (what Windows gvim does)

### Version of Vim and architecture

Vim 9.0, macOS version arm64, Included patches 1-472

### Environment

Operating system: macOS Monterey version 12.6 (21G115)
Terminal: iTerm2 (doesn't really apply, I've only seen issue in macvim GUI)
$TERM: xterm-256color
Shell: zsh 5.8.1

### How MacVim was installed

Homebrew cask

### Vim configuration where issue is reproducable

Reproducable on almost-clean vim config (with a couple commands as described)

### Issue has been tested with given configuration

- [X] by running MacVim.app from GUI macOS interface
- [X] by running vim/gvim/etc installed by MacVim
- [ ] by running other versions of vim (e.g. /usr/bin/vim)

### Issue has been tested with no configuration

- [X] by running `mvim --clean` (or `gvim`, supplied by MacVim distribution)
- ~~[ ] by running `vim --clean` (in terminal, supplied by MacVim distribution)~~
- ~~[ ] by running `vim --clean` (in terminal, other suppliers, e.g. /usr/bin/vim)~~ N/A - only applies if GUI scrollbar visible

### Other conditions

- [ ] The both Homebrew packages "vim" and "macvim" are installed
Macvim cask is installed, macvim and vim formulae are not

**Comment by eirnym** [comment]
Thank you for an issue. 

I took it as a natural tread-off for  a pixel-by-pixel resisability (as the opposite to character-by character)

In MacVim I personally don't use tabline (and Vim tabs in general) as I prefer macOS-style ones if I use tabs. 

As far as I see what's happening, it matches normal background style, so I'd propose to match this square to _tabline_ background style rather than general one.

**Comment by adinklotz** [comment]
Yeah I don't really use Vim tabs in MacVim, or in general, but I really like the way vim-airline [hijacks the tabline to show buffers](https://github.com/vim-airline/vim-airline#smarter-tab-line), which does require using the Vim text tabline.

For reference, here's how it looks in Windows's Gvim:
![Screenshot 2022-09-22 100145](https://user-images.githubusercontent.com/16441122/191768300-2af77ec3-8b70-45ae-a5a4-16c92f2993da.png)
(of course pretty ugly in that default color scheme, but nicer than the black square IMO)

Obviously this is a very minor issue, it just disproportionately bugs me to have MacVim messing up the look of my *artisanally crafted vimrc* haha

**Comment by ychin** [comment]
This is a bug. I'll try to fix it next release. Incidentally there is a new overlay scroll bar feature coming "soon" which would hopefully mean you don't have to use the old school scroll bars anymore.

And there is also a pull request to revamp of the MacVim tabs that I need to merge but I imagine that won't apply to you if you are going to turn GUI tabs off anyway.

**Comment by ychin** [comment]
I think this was fixed a while ago to use the colorscheme background. Didn't get around to close it.
